pyiges/iges.py

```diff
@@ -352,11 +374,23 @@ def _read(self, filename):
                         first_param_line = True
                         param_string = param_string.strip()[:-1]
                         parameters = param_string.split(param_sep)
-                        entity_list[pointer_dict[directory_pointer]]._add_parameters(parameters)
```

Comment:
Here the per-entity discarding machanism is implemented.

Note that I decided not to follow Your suggestion of adding the try-catch-logic to the `Entity` base class. The reason is, that the success needs to be communicated to the parser and an error message must be created anyways, which however needs also details from the parser side. So I found it a better separation of concerns keeping all parsing logic on the parser side instead of spreading it partly to the entity side. Regarding the parsing step, the entity objects are quite passive with very few behavior while the parsing class is full of the parsing logic and I found it cleaner to preserve this loose coupling.

Let me know in case You don't agree.

Reply:
That works.